src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -537,6 +544,78 @@ func (c controller) addRouterCAToClusterCA() bool {
 
 }
 
+func (c controller) getProgressingOLMOperators() ([]models.MonitoredOperator, error) {
+	ret := make([]models.MonitoredOperator, 0)
+	operators, err := c.ic.GetClusterMonitoredOLMOperators(context.TODO(), c.ClusterID)
+	if err != nil {
+		c.log.WithError(err).Warningf("Failed to connect to assisted service")
+		return ret, err
+	}
+	for _, operator := range operators {
+		monitoredOperator, err := c.ic.GetClusterMonitoredOperator(context.TODO(), c.ClusterID, operator.Name)
+		if err != nil {
+			return ret, err
+		}
+		if monitoredOperator.Status != models.OperatorStatusAvailable && monitoredOperator.Status != models.OperatorStatusFailed {
+			ret = append(ret, operator)
+		}
+	}
+	return ret, nil
+}
+
+func (c controller) getMaximumOLMTimeout() time.Duration {
+	operators, err := c.ic.GetClusterMonitoredOLMOperators(context.TODO(), c.ClusterID)
+	if err != nil {
+		c.log.WithError(err).Warningf("Failed to connect to assisted service")
+		return GeneralWaitInterval
+	}
+
+	timeout := float64(0)
+	for _, operator := range operators {
+		timeout = math.Max(float64(operator.TimeoutSeconds), float64(timeout))
+	}
+
+	return time.Duration(timeout)
+}
+
+// waitForOLMOperators wait until all OLM monitored operators are available or failed.
+func (c controller) waitForOLMOperators() bool {
+	c.log.Infof("Checking OLM operators")
+	operators, _ := c.getProgressingOLMOperators()
+	if len(operators) == 0 {
+		return true
+	}
+	for _, operator := range operators {
+		operatorName, operatorNamespace, err := utils.GetOLMNameAndNamespace(operator.Name)
+		if err != nil {
+			c.log.WithError(err).Warnf("Failed to get subscription name %s", operator.Name)
+			continue
+		}
+
+		csvName, err := c.kc.GetCSVFromSubscription(operatorNamespace, operatorName)
+		if err != nil {
+			c.log.WithError(err).Warnf("Failed to get subscription of operator %s", operatorName)
+			continue
+		}
+
+		csv, err := c.kc.GetCSV(csvName, operatorNamespace)
+		if err != nil {
+			c.log.WithError(err).Warnf("Failed to get %s", operator.Name)
+			continue
+		}
```

Comment:
From `assisted-installer` point of view we know only some abstract kind of name like: `cnv`, `ocs` or `lso`. But we have no idea what's subscription name or the CSV name. So we have a mapping method which maps names like `cnv`, `lso`, to the [subscription name](https://github.com/openshift/assisted-service/blob/master/internal/operators/lso/utils.go#L16). And from the subscription name we can latter know the actual CSV name. I think we should extract the subscription name to some public API, or at least don't store it in internal package in `asssisted-service`, so it can be used in `assisted-installer` code, and we don't have to duplicate it `GetSubscriptionNameAndNamespace` method.

Reply:
Same API as from the other comment https://github.com/openshift/assisted-installer/pull/214#discussion_r590033931